keras_retinanet/losses.py

```diff
@@ -64,16 +63,15 @@ def smooth_l1(sigma=3.0):
     sigma_squared = sigma ** 2
 
     def _smooth_l1(y_true, y_pred):
-
         # separate target and state
         regression        = y_pred
         regression_target = y_true[:, :, :4]
         anchor_state      = y_true[:, :, 4]
 
-        # compute the divisor: for each image in the batch, we want the number of positive anchors
-        divisor = keras_retinanet.backend.where(keras.backend.equal(anchor_state, 1), anchor_state, keras.backend.zeros_like(anchor_state))
-        divisor   = keras.backend.sum(divisor, axis=1, keepdims=True)
-        divisor   = keras.backend.maximum(1.0, divisor)
+        # compute the divisor: for each image in the batch, we want the number of positive and negative anchors
+        divisor = keras_retinanet.backend.where(keras.backend.not_equal(anchor_state, -1), keras.backend.ones_like(anchor_state), keras.backend.zeros_like(anchor_state))
```

Comment:
I have doubts regarding this change. This will have the effect of dividing the regression loss by (n_pos_anchors + n_neg_anchors). But only the positive anchors contribute to the loss. This effectively means we are no longer taking the mean regression loss.

Reply:
I see your point. The change was made because it seems to be what is happening in `py-faster-rcnn` too. Do you think it's better to diverge from `py-faster-rcnn` here? It does mean the regression loss is relative high (around 0.9 when converged). Other than that it shouldn't really matter.